src/openms/source/ANALYSIS/DECHARGING/MetaboliteFeatureDeconvolution.cpp

```diff
@@ -0,0 +1,1209 @@
+      s_comp(s), idx_cp(idx), side_cp(side) {}
+
+    // Copy C'tor
+    CmpInfo_(const CmpInfo_& rhs) :
+      s_comp(rhs.s_comp), idx_cp(rhs.idx_cp), side_cp(rhs.side_cp) {}
+
+    // Assignment
+    CmpInfo_& operator=(const CmpInfo_& rhs)
+    {
+      if (&rhs == this) return *this;
+
+      s_comp = rhs.s_comp;
+      idx_cp = rhs.idx_cp;
+      side_cp = rhs.side_cp;
+      return *this;
+    }
+
+    // Comparator
+    bool operator<(const CmpInfo_& other) const
+    {
+      if (s_comp < other.s_comp) return true; else return false;
+    }
+
+    bool operator==(const CmpInfo_& other) const
+    {
+      if (s_comp == other.s_comp) return true; else return false;
+    }
+
+  };
+
+  MetaboliteFeatureDeconvolution::MetaboliteFeatureDeconvolution() :
+    DefaultParamHandler("MetaboliteFeatureDeconvolution"),
+    potential_adducts_(),
+    map_label_(),
+    map_label_inverse_(),
+    enable_intensity_filter_(false)
+  {
+    defaults_.setValue("charge_min", 1, "Minimal possible charge");
+    defaults_.setValue("charge_max", 3, "Maximal possible charge");
+    defaults_.setValue("charge_span_max", 3, "Maximal range of charges for a single analyte, i.e. observing q1=[5,6,7] implies span=3. Setting this to 1 will only find adduct variants of the same charge");
+    defaults_.setMinInt("charge_span_max", 1); // will only find adduct variants of the same charge
+    defaults_.setValue("q_try", "feature", "Try different values of charge for each feature according to the above settings ('heuristic' [does not test all charges, just the likely ones] or 'all' ), or leave feature charge untouched ('feature').");
+    defaults_.setValidStrings("q_try", ListUtils::create<String>("feature,heuristic,all"));
+
+    defaults_.setValue("retention_max_diff", 1.0, "Maximum allowed RT difference between any two features if their relation shall be determined");
+    defaults_.setValue("retention_max_diff_local", 1.0, "Maximum allowed RT difference between between two co-features, after adduct shifts have been accounted for (if you do not have any adduct shifts, this value should be equal to 'retention_max_diff', otherwise it should be smaller!)");
+    /// TODO should be m/z ppm?!
+    defaults_.setValue("mass_max_diff", 0.05, "Maximum allowed mass difference [in Th] for a single feature.");
+    // Na+:0.1 , (2)H4H-4:0.1:-2:heavy
+    defaults_.setValue("potential_adducts", ListUtils::create<String>("H+:0.9"), "Adducts used to explain mass differences in format: 'Element(+)*:Probability[:RTShift[:Label]]', i.e. the number of '+' indicate the charge, e.g. 'Ca++:0.5' indicates +2. Probabilites have to be in (0,1]. RTShift param is optional and indicates the expected RT shift caused by this adduct, e.g. '(2)H4H-4:1:-3' indicates a 4 deuterium label, which causes early elution by 3 seconds. As a fourth parameter you can add a label which is tagged on every feature which has this adduct. This also determines the map number in the consensus file.");
```

Comment:
wouldnt it make sense to have a few good adducts as defaults here (in addition to H+) ?